web/pgadmin/preferences/__init__.py

```diff
@@ -269,6 +276,12 @@ def save():
                             samesite=config.SESSION_COOKIE_SAMESITE,
                             **domain)
 
+    # if layout is changed from 'Workspace' to 'Classic', disconnect all
+    # servers.
+    if has_layout_changed:
+        disconnect_from_all_servers()
```

Comment:
Following the `separation of concerns` principle, this can be taken care in a separate API call after successful pref save.

Reply:
Fixed